client/blocks/stats-navigation/index.js

```diff
@@ -277,15 +277,34 @@ class StatsNavigation extends Component {
 			!! ( statsAdminVersion && version_compare( statsAdminVersion, '0.9.0-alpha', '>=' ) );
 
 		const shouldRenderModuleToggler =
-			! isLegacy &&
 			isModuleSettingsSupported &&
 			AVAILABLE_PAGE_MODULES[ this.props.selectedItem ] &&
 			! hideModuleSettings &&
 			! gatedTrafficPage;
 
 		// @TODO: Add loading status of modules settings to avoid toggling modules before they are loaded.
 
-		const validNavItems = Object.keys( navItems ).filter( this.isValidItem );
+		/** @type {(keyof typeof allNavItems)[]} */
```

Comment:
```suggestion
		/** @type {Array<keyof typeof allNavItems>} Array of valid navigation item keys */
```
Could we provide more detailed annotations if we tend to?